### PR TITLE
TikZ export of DiffTrees

### PR DESCRIPTION
**Pull request by ibbem**

### CODE REVIEW

**Review by pmbittner — COMMENTED** [review]
Changes look very good. Proceed as discussed.

File: src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/GraphvizExporter.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+     * Replaces all special characters with uninterpreted escape codes.
+     *
+     * The Graphviz parser for strings interprets some characters specially. The result of this
+     * function can be used in Graphviz strings (surrounded by double quotes) resulting in all
+     * characters appearing literally in the output.
+     *
+     * Note that some backends of Graphviz may still interpret some strings specially, most
+     * commonly strings containing HTML tags.
+     */
+    private String escape(List<String> label) {
```
Comment (pmbittner) [inline src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/GraphvizExporter.java:141]: Please add documentation for the callee. What is `label` supposed to be? Is it a set of lines?

File: src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/TikzExporter.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        var output = new PrintStream(destination);
+        output.println("\\begin{tikzpicture}");
+
+        // Convert the layout information received by Graphviz to coordinates used by TikZ.
+        try (var graphizOutput =
+                new BufferedReader(
+                    new InputStreamReader(
+                        new GraphvizExporter(format)
+                            .computeGraphvizLayout(
+                                diffTree,
```
Comment (pmbittner) [inline src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/TikzExporter.java:69]: Is this nesting safe? Are all readers closed upon an exception?

File: src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/GraphvizExporter.java (hunk 1 above)
Comment (pmbittner) [reply src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/GraphvizExporter.java:141]: Also make this static as it should not and does not rely on object state.

**Review by pmbittner — CHANGES_REQUESTED** [review]
Only minor changes. Looks very good!

File: src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/TikzExporter.java (hunk 2 above)
Comment (ibbem) [reply src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/TikzExporter.java]: You're wright, some intermediate constructor might throw an exception (e.g. `OutOfMemoryError`) which would result in possible resource leaks.

**Comment by ibbem** [comment]
All requested changes are done now.

**Review by pmbittner — APPROVED** [review]

**Pull request merged**